DAOS-17359 dfuse: Add special file for log mask

jolivier23 opened the pull request:
It may be useful to interact with dfuse at runtime to perform certain tasks. This adds a generic
command parser capability and two initial commands as a reference.

### log command

Changing the log mask at runtime can be useful to debug certain runtime behaviors without
overwhelming the default log.  The log command allows us to do this.

```shell
cat << EOF > /mnt/dfuse/.dfuse_ctrl
log
log_mask=debug
streams=all
EOF
```
To see any errors from the command (or the output), cat the file
```shell
cat /mnt/dfuse/.dfuse_ctrl
```

### stats command

```shell
echo "stats" > /mnt/dfuse/.dfuse_ctrl; cat /mnt/dfuse/.dfuse_ctrl
```

This will produce the same output as `daos fs query /mnt/dfuse`

The daos command requires the client to be installed and this was not always the case when dfuse is mounted in a sidecar container.

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

### Steps for the author:

* [ ] Commit message follows the [guidelines](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Appropriate [Features or Test-tag](https://daosio.atlassian.net/wiki/spaces/DC/pages/10984259629/Test+Tags) pragmas were used.
* [ ] Appropriate [Functional Test Stages](https://daosio.atlassian.net/wiki/spaces/DC/pages/12147556353/CI+Functional+Test+Stages) were run.
* [ ] At least two positive code reviews including at least one code owner from each category referenced in the PR.
* [ ] Testing is complete. If necessary, forced-landing label added and a reason added in a comment.

#### After all prior steps are complete:
* [ ] Gatekeeper requested (daos-gatekeeper added as a reviewer).

jolivier23 committed:
DAOS-17359 dfuse: Add special file log mask

Sometimes it's useful to see what is happening in the client at runtime
after some condition has been hit.  Add a special file named
.dfuse_log_ctrl that is a write only file for setting log mask.

$ cat << EOF > /mnt/dfuse/.dfuse_log_ctrl
log_mask=debug
streams=all
EOF

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

jolivier23 committed:
Add a test and move stuff to gurt

NLT will have some new stable warnings

Allow-unstable-test: true

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16182/4/execution/node/292/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16182/4/execution/node/268/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.5 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16182/4/execution/node/244/log

daosbuild1 commented:
Test stage NLT on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-16182/5/testReport/

jolivier23 committed:
Merge remote-tracking branch 'comm/master' into jvolivie/dfuse_log

jolivier23 committed:
Change file name to .dfuse_ctrl

This opens possibility of using it for other purposes.
Allow-unstable-test: true

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

daosbuild1 commented:
Test stage NLT on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-16182/6/testReport/

jolivier23 commented:
@mchaarawi any thoughts on the general concept here?

mchaarawi commented:
> @mchaarawi any thoughts on the general concept here?

not opposed to it. my question though is what is better, this approach or just add an ioctl and update a daos tool to pass the logging needed?
the latter does not need all those special handling of the ctrl file.

jolivier23 commented:
> > @mchaarawi any thoughts on the general concept here?
> 
> not opposed to it. my question though is what is better, this approach or just add an ioctl and update a daos tool to pass the logging needed? the latter does not need all those special handling of the ctrl file.

The problem is that it's not always convenient to get the daos tool.   Most of our GKE customers just have a mount point and no control over it or daos client installed.

Though maybe we could provide some means that is relatively simple (I did previously write a python script to do an ioctl)

mchaarawi commented:
> > > @mchaarawi any thoughts on the general concept here?
> > 
> > 
> > not opposed to it. my question though is what is better, this approach or just add an ioctl and update a daos tool to pass the logging needed? the latter does not need all those special handling of the ctrl file.
> 
> The problem is that it's not always convenient to get the daos tool. Most of our GKE customers just have a mount point and no control over it or daos client installed.
> 
> Though maybe we could provide some means that is relatively simple (I did previously write a python script to do an ioctl)

not insisting on that really, just thought it would be cleaner?
but if there is a usage issue with that, then this is fine too.

jolivier23 committed:
Merge remote-tracking branch 'comm/master' into jvolivie/dfuse_log

jolivier23 committed:
Try without changing logleaks default

Allow-unstable-test: true

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

daltonbohning committed:
DAOS-17406 cq: use cancel-in-progress for more workflows (#16244)

So previous runs are cancelled on new pushes.

Signed-off-by: Dalton Bohning <dalton.bohning@hpe.com>

daosbuild1 commented:
Test stage NLT on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-16182/7/testReport/

jolivier23 committed:
No leak

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

jolivier23 committed:
Allow-unstable-test: true

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

daosbuild1 commented:
Test stage NLT on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-16182/9/testReport/

jolivier23 committed:
Try again

Allow-unstable-test: true

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

daosbuild1 commented:
Test stage NLT on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-16182/10/testReport/

jolivier23 committed:
DAOS-17404 ci: Enable plugin for finding branches (#16234)

When we are using a repo for other than GitHub, things
may not be in proper places for the defaults to work.
This enables running plugin scripts to get extra
branch names

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

Commit:
DAOS-623 build: Fix protobuf generator Makefile (#16257)

Also regenerate protobufs so that they've all been done with
the same versions of the tools.

Signed-off-by: Michael MacDonald <mjmac@google.com>

daltonbohning committed:
DAOS-17400 test: add feature tags to io_sys_admin.py (#16231)

And to critical_integration.py since regressions have been missed multiple times.

Signed-off-by: Dalton Bohning <dalton.bohning@hpe.com>

daltonbohning committed:
DAOS-17423 test: disable test_dfuse_find_perf in CI (#16262)

test_dfuse_find_perf is a manual test that requires Lustre and we do not have Lustre so disable it.

Signed-off-by: Dalton Bohning <dalton.bohning@hpe.com>

jolivier23 committed:
Revert "Try without changing logleaks default"

This reverts commit eef164fb338249152d4081ff79781282dbf6cf7e.

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

jolivier23 committed:
Allow-unstable-test: true

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

daosbuild1 commented:
Test stage NLT on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-16182/11/testReport/

daosbuild1 commented:
Test stage NLT on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-16182/12/testReport/

Nasf-Fan committed:
DAOS-17353 object: reduce noisy warning for DTX refresh (#16184)

When process IO on DTX non-leader target, if hit prepared DTX,
related retry logic may generate warning message for reminder
before refresh related DTX. That is not fatal but some noisy.

This patch controls such warning message rate to avoid flood
in log: one warning message per about 10 seconds.

Signed-off-by: Fan Yong <fan.yong@hpe.com>

jolivier23 committed:
Fix nlt test

Allow-unstable-test: true

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

daosbuild1 commented:
Test stage NLT on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16182/13/execution/node/735/log

jolivier23 committed:
Allow-unstable-test: true

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

daosbuild1 commented:
Test stage NLT on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16182/14/execution/node/734/log

liw committed:
DAOS-17405 cart: Add env var SWIM_SUBGROUP_SIZE (#16228)

Make the number of iping targets, called the subgroup size, tunable by
new environment variable SWIM_SUBGROUP_SIZE.

Log new SWIM suspicions at INFO level, including their origins. This may
help debugging why a DEAD event happens.

Signed-off-by: Li Wei <liwei@hpe.com>

jolivier23 committed:
Allow-unstable-test: true

Trying to get errors down to just my test

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

daosbuild1 commented:
Test stage NLT on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16182/16/execution/node/825/log

jolivier23 committed:
Disable log test for logctrl test

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

daosbuild1 commented:
Test stage NLT on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-16182/17/testReport/

NiuYawei committed:
DAOS-17345 vos: reduce sys space reserving (#16215)

Keep reserving 5% of total_scm_size for anti-fragmentation, but reduce
the min reservation from 2GB to 600MB, reduce the max reservation from
10GB to 6GB.

Signed-off-by: Niu Yawei <yawei.niu@hpe.com>

sherintg committed:
DAOS-17312: Make bucket memory allocator the default for md_on_ssd (#16132)

Moving forward, the bucket memory allocator will be the default for
new pools created in md_on_ssd mode. This also applies to pools with
a 100% memory-to-meta ratio. Existing pools that were created using
the phase1 allocator will continue to use it.

Signed-off-by: Sherin T George <sherin-t.george@hpe.com>

Nasf-Fan committed:
DAOS-17347 dtx: retry DTX resync for recoverable network failure (#16173)

* DAOS-17347 dtx: retry DTX resync for recoverable network failure

If the engine on which some DTX leader reside restart, then former
in-DRAM committable DTX status will be lost. When (re)open related
container, DTX resync will be triggered to resync and commit those
former committable DTX entries. But during DTX resync, it may hit
recoverable netwrok failure, such as "-DER_OOG" since lower layer
network membership may be not ready yet. Under such case, wen need
to retry resync for related DTX; otherwise related DTX will become
non-committable status globally as to misguide subsequent IO logic
to give wrong visibility decision.

On the other hand, if some non-leader triggers DTX refresh before
DTX resync completed on the leader, it will get "-DER_INPROGRESS"
and needs to retry.

Signed-off-by: Fan Yong <fan.yong@hpe.com>

Commit:
DAOS-17278 build: Update golang.org/x/net to v0.37.0 (#16128)

Address a CVE. Requires an update to the minimum toolchain
version to 1.23.0. If the build toolchain is at least 1.21,
then a newer toolchain will automatically be downloaded.

* Add "toolchain" to spellcheck dictionary

Co-authored-by: Kris Jacque <kris.jacque@hpe.com>
Signed-off-by: Kris Jacque <kris.jacque@hpe.com>
Signed-off-by: Michael MacDonald <mjmac@google.com>

jolivier23 committed:
No need to restore log mask

Not sure why this is not already working

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

jolivier23 committed:
Empty commit

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

jolivier23 committed:
Merge remote-tracking branch 'comm/master' into jvolivie/dfuse_log

jolivier23 committed:
Not sure what happened here but trying to fix it

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

jolivier23 committed:
Remove duplicate test

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

daosbuild1 commented:
Test stage NLT on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-16182/20/testReport/

jolivier23 committed:
Merge remote-tracking branch 'comm/master' into jvolivie/dfuse_log

jolivier23 committed:
Change how this is done to make it more generic

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

jolivier23 committed:
Fix a couple of issues

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

jolivier23 committed:
Rearrange a bit

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

daosbuild3 commented:
Test stage NLT on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16182/8/testReport/

jolivier23 committed:
Merge remote-tracking branch 'comm/master' into jvolivie/dfuse_log

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

jolivier23 committed:
Fix some stuff

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

jolivier23 committed:
Modify test

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

jolivier23 committed:
Fix linting issues

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

jolivier23 committed:
Fix more linting stuff

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

jolivier23 committed:
Merge remote-tracking branch 'comm/master' into jvolivie/dfuse_log

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

daosbuild3 commented:
Test stage NLT on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16182/12/testReport/

jolivier23 committed:
Merge remote-tracking branch 'comm/master' into jvolivie/dfuse_log

Change-Id: I573327e9274870e4c9f9e7602097d888bd6f5e33

jolivier23 committed:
Just merge with latest

Change-Id: I7815777a0bc1b8b0bbcade4df5cc5566b9552947
Signed-off-by: Jeff Olivier <jeffolivier@google.com>

daosbuild3 commented:
Test stage NLT on EL 8.8 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos//view/change-requests/job/PR-16182/13/testReport/
